File: Aptech/C language/C/C/Others/CALC_POWER.c
#include <stdio.h>
// calculate power using loop
int main()
{
    int p;
    float base, power, power2, prod_index = 1.0;
    printf("\n Enter base & power : ");
    scanf("%f %f", &base, &power);
    if (power >= 1)
    {
        for (p = 1; p <= power; p++)
        {
            prod_index *= base;
        }
    }
    else if (power < 0)
    {
        power2 = -power;
        for (p = 1; p <= power2; p++)
        {
            prod_index *= base;
        }
        prod_index = (float)(1 / prod_index);
    }
    else if (power == 0)
    {
        prod_index = 1;
    }
    printf("\n The result = %.2f", prod_index);

    return 0;
}

File: JAVA/Example_java/CalcPower.java
import java.util.*;

public class CalcPower
{
   public static void main(String args[])
   {
	
      Scanner sc=new Scanner(System.in);
      double b,p,r;
      System.out.println("Enter the base ");
      b=sc.nextDouble();
      System.out.println("Enter the power ");
      p=sc.nextDouble();
      r=powerFunc(b,p);
      System.out.println("The result:=   "+r);

   }
   public static double powerFunc(double base,double power)
   {        
       double r,p;
       if(power==0.0)
         r=1.0;             
       else if(power<0.0)
       {
          p=-power;
          r=(double)(1/(double)(base*powerFunc(base,p-1))); 
          
       } 
        else 	 	  
         r=base*powerFunc(base,power-1); 
       return r; 

   }
}
			